docs/source/quickstart.rst

```diff
@@ -0,0 +1,125 @@
+Quickstart
+==========
+
+A ConfigSpace is equal to a container holding all different kinds of hyperparameters.
+For example numerical, categorical hyperparameters and condtional hyperparameters.
+As well as ordinal hyperparameters.
+
+Tools like `SMAC3`_, `BOHB`_ or `auto-sklearn`_ can sample hyperparameter-configurations from this ConfigSpace.
+
+This simple quickstart tutorial will show you, how to set up you own ConfigSpace, and what you can realize with it.
+To accomplish this task, we need to:
+
+- Create a ConfigSpace
+- Define hyperparameters and their value ranges
+- Add the hyperparameters to the ConfigSpace
+- (Optional) Add constraints to the ConfigSpace
+
+We will show those steps in an exemplary way, by creating a ConfigSpace for a soft-margin SVM classifier and a neural network.
+
+1. Example: Basic Usage and Conditional Hyperparameter
+------------------------------------------------------
+
+Assume, we want to train a soft-margin SVM classifier with a RBF-kernel. This classifier has at least two hyperparametrs.
+A regularization constant :math:`\mathcal{C}` and a kernel hyperparameter  :math:`\gamma` with
+
+- :math:`\mathcal{C} \in` {10, 100, 1000}
+- :math:`\gamma \in` {0.1, 0.2, 0.5, 1.0}
+
+| How this classifier is implemented is not necessary in this example and thus not shown.
+
+The first step is always to create a ConfigSpace-object. All the hyperparameters and constraints will be added to this
+object.
+::
+
+   import ConfigSpace as CS
+   cs = CS.ConfigurationSpace()
+
+Now, we have to define the hyperparameters :math:`\mathcal{C}` and :math:`\gamma`. Since each hyperparameter only can
+take on values from a given subset, we use categorical-hyperparameters in this case.
+:: 
+
+   import ConfigSpace.hyperparameters as CSH
+   c = CSH.CategoricalHyperparameter(name='C', choices=[10, 100, 1000])
+   gamma = CSH.CategoricalHyperparameter(name='gamma', choices=[0.1, 0.2, 0.5, 1.0])
+
+As last step in this first example, we only need to add them to the ConfigSpace::
+
+   cs.add_hyperparameter(c)
+   cs.add_hyperparameter(gamma)
+
+And that's it.
```

Comment:
It would be good to do something with the configuration space at this point. Could you please add an example on sampling configurations from it?